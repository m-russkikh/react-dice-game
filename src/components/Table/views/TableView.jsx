import React from 'react';
import { Table } from 'reactstrap';
import md5 from 'blueimp-md5';

import { TableViewPropType } from '../propTypes';


function TableView({ history, autoBet }) {
  if (!autoBet) {
    return null;
  }

  return (
    <Table striped>
      <thead>
        <tr>
          <th>Bet number</th>
          <th>Bet amount</th>
          <th>Result</th>
          <th>Probably Fair Hash</th>
          <th>Balance</th>
        </tr>
      </thead>
      <tbody>
        { history.map(item => (
          <tr key={item.betNumber}>
            <th scope="row">{item.betNumber}</th>
            <td>{item.betAmount}</td>
            <td>{item.resultNumber} {item.win ? 'WIN' : 'LOSE'}</td>
            <td>{md5(item.resultNumber)}</td>
            <td>{item.balance}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
}

TableView.propTypes = TableViewPropType;

export default TableView;
