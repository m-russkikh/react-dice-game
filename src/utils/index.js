import { MAX_BET_NUMBER } from '../constants/diceGame';


export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getBetNumber() {
  return Math.floor(Math.random(0, 1) * MAX_BET_NUMBER) + 1;
}
