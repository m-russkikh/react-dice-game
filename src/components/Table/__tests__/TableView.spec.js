import { getShallowWrapper } from '../../../utils/testUtils';
import TableView from '../views/TableView';


describe('TableView', () => {
  describe('render', () => {
    it('should not display TableView', () => {
      const props = {
        autoBet: false,
        history: [],
      };
      const wrapper = getShallowWrapper(TableView, props);

      expect(wrapper.find('Table').exists()).toBeFalsy();
    });

    it('should display TableView', () => {
      const props = {
        autoBet: true,
        history: [{
          betNumber: 1,
          betAmount: 10,
          resultNumber: 10,
          balance: 100,
          win: true,
        }],
      };
      const wrapper = getShallowWrapper(TableView, props);

      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
