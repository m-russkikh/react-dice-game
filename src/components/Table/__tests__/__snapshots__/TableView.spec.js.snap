// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableView render should display TableView 1`] = `
"<Table striped={true} tag=\\"table\\" responsiveTag=\\"div\\">
  <thead>
    <tr>
      <th>
        Bet number
      </th>
      <th>
        Bet amount
      </th>
      <th>
        Result
      </th>
      <th>
        Probably Fair Hash
      </th>
      <th>
        Balance
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope=\\"row\\">
        1
      </th>
      <td>
        10
      </td>
      <td>
        10
         
        WIN
      </td>
      <td>
        d3d9446802a44259755d38e6d163e820
      </td>
      <td>
        100
      </td>
    </tr>
  </tbody>
</Table>"
`;
